packages/analytics/amplify_analytics_pinpoint_dart/lib/src/impl/flutter_provider_interfaces/device_context_info_provider.dart

```diff
@@ -0,0 +1,187 @@
+      appVersion: _sanitize(appVersion),
+      make: _sanitize(make),
+      model: _sanitize(model),
+      modelVersion: _sanitize(modelVersion),
+      platformVersion: _sanitize(platformVersion),
+      platform: platform,
+    );
+  }
+
+  const DeviceContextInfo._({
+    this.countryCode,
+    this.locale,
+    this.timezone,
+    this.appName,
+    this.appPackageName,
+    this.appVersion,
+    this.make,
+    this.model,
+    this.modelVersion,
+    this.platform,
+    this.platformVersion,
+  });
+
+  static final AmplifyLogger _logger =
+      AmplifyLogger.category(Category.analytics);
+
+  static const int _maxFieldLength = 50;
+
+  /// ISO 3166-1, Alpha-2, Alpha-3 country code format
+  ///
+  /// Ex: US or USA
+  ///
+  /// {@macro amplify_analytics_pinpoint_dart.device_context_info.sent_to_pinpoint_endpoint}
+  final String? countryCode;
+
+  /// IETF BCP 47 locale format
+  ///
+  /// Ex: en_US es_419 haw_US
+  ///
+  /// {@macro amplify_analytics_pinpoint_dart.device_context_info.sent_to_pinpoint_endpoint}
+  final String? locale;
+
+  /// tz database timezone name
+  ///
+  /// Due to complexities with tz database, this field is not auto-set!
+  ///
+  /// {@macro amplify_analytics_pinpoint_dart.device_context_info.sent_to_pinpoint_endpoint}
+  // TODO(fjnoyp): this is currently ignored
+  final String? timezone;
+
+  /// Maps to 'AppTitle' in PinpointEvent.
+  ///
+  /// {@macro amplify_analytics_pinpoint_dart.device_context_info.sent_to_pinpoint_event}
+  final String? appName;
+
+  /// The package name of the app
+  ///
+  /// {@macro amplify_analytics_pinpoint_dart.device_context_info.sent_to_pinpoint_event}
+  final String? appPackageName;
+
+  /// Version of the app
+  ///
+  /// {@macro amplify_analytics_pinpoint_dart.device_context_info.sent_to_pinpoint_endpoint}
+  final String? appVersion;
+
+  /// Manufacturer such as Apple or Samsung
+  ///
+  /// {@macro amplify_analytics_pinpoint_dart.device_context_info.sent_to_pinpoint_endpoint}
+  final String? make;
+
+  /// Model name or number of device
+  ///
+  /// Ex: iPhone SM-G900F
+  /// {@macro amplify_analytics_pinpoint_dart.device_context_info.sent_to_pinpoint_endpoint}
+  final String? model;
+
+  /// Model version
+  ///
+  /// {@macro amplify_analytics_pinpoint_dart.device_context_info.sent_to_pinpoint_endpoint}
+  final String? modelVersion;
+
+  /// Version of the platform
+  ///
+  /// {@macro amplify_analytics_pinpoint_dart.device_context_info.sent_to_pinpoint_endpoint}
+  final String? platformVersion;
+
+  /// Platform of the device
+  ///
+  /// {@macro amplify_analytics_pinpoint_dart.device_context_info.sent_to_pinpoint_endpoint}
+  final DevicePlatform? platform;
+
+  static String? _sanitize(String? field) {
+    if (field == null || field.length <= _maxFieldLength) return field;
+
+    final newField = field.substring(0, _maxFieldLength);
+    _logger.warn(
+      'Warning - DeviceContextInfoProvider '
```

Comment:
Can remove this line. Warning is already a given since you're calling `.warn` and I left a comment above to put `DeviceContextInfoProvider` in the logger name